lib/components/base/Checkbox.vue

```diff
@@ -23,45 +23,36 @@
     <slot v-else />
   </div>
 </template>
-<script setup>
+<script setup lang="ts">
 import { CheckIcon, DropdownIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  props: {
-    label: {
-      type: String,
-      default: '',
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    description: {
-      type: String,
-      default: '',
-    },
-    modelValue: Boolean,
-    clickEvent: {
-      type: Function,
-      default: () => {},
-    },
-    collapsingToggleStyle: {
-      type: Boolean,
-      default: false,
-    },
-  },
-  emits: ['update:modelValue'],
-  methods: {
-    toggle() {
-      if (!this.disabled) {
-        this.$emit('update:modelValue', !this.modelValue)
-      }
-    },
-  },
-})
+
+const emit = defineEmits<{
+  'update:modelValue': [boolean]
+}>()
+
+const props = withDefaults(
+  defineProps<{
+    label: string
+    disabled: boolean
+    description: string
+    modelValue: boolean
+    clickEvent: () => void
+    collapsingToggleStyle: boolean
+  }>(),
+  {
+    label: '',
+    disabled: false,
+    description: '',
+    clickEvent: () => {},
+    collapsingToggleStyle: false,
+  }
+)
```

Comment:
```suggestion
const props = withDefaults(
  defineProps<{
    label: string
    disabled?: boolean
    description: string
    modelValue: boolean
    clickEvent?: () => void
    collapsingToggleStyle?: boolean
  }>(),
  {
    label: '',
    disabled: false,
    description: '',
    modelValue: false,
    clickEvent: () => {},
    collapsingToggleStyle: false,
  }
)
```

We can probably fix a mistake where some props would be marked as required, even though they are in fact optional, like `disabled`, `clickEvent` [^1], and `collapsingToggleStyle`. Omission of those props won't change anything, as we have the defaults; unlike things like lack of `modelValue`, `label`, or `description`, which is likely to be a mistake when it comes to this component.

Also, it's probably a good idea to also have a default for `modelValue`: omitted props without a default are initialised to `undefined`, which is not great. Even though `undefined` wouldn't break this component, consistency is nice.

[^1]: I wonder why we aren't using emit tbf, but introducing a breaking change is not a scope of this PR.

Reply:
Yeah, it's better ^^.